kasa/tests/test_protocol.py

```diff
@@ -129,6 +129,36 @@ def aio_mock_writer(_, __):
         assert "success" not in caplog.text
 
 
+@pytest.mark.parametrize("custom_port", [123, 9999])
+async def test_protocol_custom_port(mocker, custom_port):
+    encrypted = TPLinkSmartHomeProtocol.encrypt('{"great":"success"}')[
+        TPLinkSmartHomeProtocol.BLOCK_SIZE :
+    ]
+
+    async def _mock_read(byte_count):
+        nonlocal encrypted
+        if byte_count == TPLinkSmartHomeProtocol.BLOCK_SIZE:
+            return struct.pack(">I", len(encrypted))
+        if byte_count == len(encrypted):
+            return encrypted
+        raise ValueError(f"No mock for {byte_count}")
+
+    def aio_mock_writer(_, port):
+        reader = mocker.patch("asyncio.StreamReader")
+        writer = mocker.patch("asyncio.StreamWriter")
+        assert port == custom_port
+        mocker.patch.object(reader, "readexactly", _mock_read)
+        return reader, writer
+
+    if custom_port == 9999:
+        protocol = TPLinkSmartHomeProtocol("127.0.0.1")
+    else:
+        protocol = TPLinkSmartHomeProtocol("127.0.0.1", port=custom_port)
```

Comment:
```suggestion
    protocol = TPLinkSmartHomeProtocol("127.0.0.1", port=custom_port)
```
This and checking that the default port is used if the `custom_port` is `None` should be enough here.

Reply:
Updated.